### PR TITLE
Chande Variable Index Dynamic Average (VIDYA) Indicator

### LINKED ISSUE #4633: Variable Index Dynamic Average (VIDYA) Indicator
#### Expected Behavior
Support Variable Index Dynamic Average (VIDYA) indicator

#### Actual Behavior
Not supported

#### Potential Solution
Implements VIDYA ([formula](http://www.fxcorporate.com/help/MS/NOTFIFO/i_Vidya.html))

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by MysticCodolic**
#### Description

Added C# Code for Chande Variable Index Dynamic Average (VIDYA) Indicator.
Applied from 
[Indicator Source](https://www.fxcorporate.com/help/MS/NOTFIFO/i_Vidya.html)

#### Related Issue

Close #4633 
#### How Has This Been Tested?

Test data taken from python script and cross-checked with python package
[py package](https://pypi.org/project/technical/)
[data](https://github.com/QuantConnect/Lean/raw/master/Data/equity/usa/daily)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by Marinovsky — COMMENTED** [review]
Nice work! 🚀 
Leaving a couple of comments:

File: Indicators/ChandeVariableIndexDynamicAverage.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    /// This indicator computes the n-period adaptive weighted moving average indicator.
+    /// VIDYAi = Pricei x F x ABS(CMOi) + VIDYAi-1 x (1 - F x ABS(CMOi))
+    /// where:
+    /// VIDYAi - is the value of the current period.
+    /// Pricei - is the source price of the period being calculated.
+    /// F = 2/(Period_EMA+1) - is a smoothing factor.
+    /// ABS(CMOi) - is the absolute current value of CMO.
+    /// VIDYAi-1 - is the value of the period immediately preceding the period being calculated.
+    /// </summary>
+    public class ChandeVariableIndexDynamicAverage : WindowIndicator<IndicatorDataPoint>, IIndicatorWarmUpPeriodProvider
```
Comment (Marinovsky) [inline Indicators/ChandeVariableIndexDynamicAverage.cs:30]: Could you please rename it to `VariableIndexDynamicAverage`?

Comment (MysticCodolic) [reply Indicators/ChandeVariableIndexDynamicAverage.cs:30]: Sure👍

File: Algorithm/QCAlgorithm.Indicators.cs [hunk 2]
```diff
@@ -456,6 +456,24 @@ public ChaikinMoneyFlow CMF(Symbol symbol, int period, Resolution? resolution =
...
+        /// <summary>
+        /// Creates a new Chande's Variable Index Dynamic Average indicator.
+        /// </summary>
+        /// <param name="symbol">The symbol whose VIDYA we want</param>
+        /// <param name="period">The period over which to compute the VIDYA</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to the Value property of BaseData (x => x.Value)</param>
+        /// <returns>The VariableIndexDynamicAverage indicator for the requested symbol over the specified period</returns>
+        [DocumentationAttribute(Indicators)]
+        public VariableIndexDynamicAverage VIDYA(Symbol symbol, int period, Resolution? resolution = null, Func<IBaseData, decimal> selector = null)
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Indicators.cs:468]: The position of the indicators in this file are alphabetical, will need to moves this helper method to respect this

File: Indicators/VariableIndexDynamicAverage.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples > Period;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod => Period + 1;
```
Comment (Martin-Molinero) [inline Indicators/VariableIndexDynamicAverage.cs:65]: Missing to add the `override` 👍

File: Indicators/VariableIndexDynamicAverage.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
+        {
+            _CMO.Update(input);
+            if (!IsReady)
+            {
+                _vidya = input.Value;
+                return 0m;
+            }
+            decimal absCMO = Math.Abs(_CMO.Current.Value / 100);
```
Comment (Martin-Molinero) [inline Indicators/VariableIndexDynamicAverage.cs:81]: > decimal 

nit - should be `var`

**Review by Martin-Molinero — APPROVED** [review]
Thank you 👍  Leaving minor comments

**Comment by MysticCodolic** [comment]
**Workflow Tests Error: VIDYA(9)**
Failed ComparesAgainstExternalData [5 ms]
  Error Message:
     Expected: 2366326.3810000001d +/- 0.001d
  But was:  2337590.6629318395d
  Off by:   28735.718068160582d

**Tests Successful in Local: VIDYA(9)**

image


Please let me know if you have any suggestions for resolving this.

**Comment by MysticCodolic** [comment]
> **Workflow Tests Error: VIDYA(9)** Failed ComparesAgainstExternalData [5 ms] Error Message: Expected: 2366326.3810000001d +/- 0.001d But was: 2337590.6629318395d Off by: 28735.718068160582d
> 
> **Tests Successful in Local: VIDYA(9)**
> 
> image
> 
> Please let me know if you have any suggestions for resolving this.

I found that the workflow didn't recognize the UK time format. So, I have updated the test file to the US time standard. Tests should work now.
Thanks

**Pull request merged**